src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -32,6 +32,11 @@ namespace Iot.Device.Arduino
     /// </summary>
     public class ArduinoBoard : Board.Board, IDisposable
     {
+        private readonly List<SupportedMode> _knownSupportedModes = new List<SupportedMode>();
+        private readonly List<ExtendedCommandHandler> _extendedCommandHandlers = new List<ExtendedCommandHandler>();
+        private readonly ReaderWriterLockSlim _commandHandlersLock =
+            new ReaderWriterLockSlim(LockRecursionPolicy.SupportsRecursion);
```

Comment:
Why do we need to support recursion here? When using this type you want to be very careful if you allow recursion as that can create deadlocks. In general it is better to use a NoRecursion lock unless you absolutely need it.

Reply:
I generally see it the other way round: When recursion is allowed, calling other private functions withhin the same class, which in turn take the lock as well, causes no problems. Since there are no callbacks or other outgoing calls within a lock, deadlocks cannot happen here.

Reply:
If you check the documentation of the type, it tells you to try to avoid allowin recursion unless necessary. Anyway, my comment is: if you need it, then its fine, if not, better to remove it.